Prefix is doubled in cmake INTERFACE_INCLUDE_DIRECTORIES

On NixOS, we are having trouble with the file written to `$prefix/share/c++utilities/cmake/c++utilitiesTargets.cmake`. See this output in that file:

```
set_target_properties(c++utilities PROPERTIES
  INTERFACE_COMPILE_DEFINITIONS "CPP_UTILITIES_USE_NATIVE_FILE_BUFFER;CPP_UTILITIES_STATIC"
  INTERFACE_INCLUDE_DIRECTORIES "/nix/store/196qm6alw4a5v4ifg2h7n7yn3c2gry5f-cpp-utilities-5.2.0//nix/store/196qm6alw4a5v4ifg2h7n7yn3c2gry5f-cpp-utilities-5.2.0/include"
)
```

As you can see, the prefix is written twice from some reason. This issue makes the build for qtutilities and syncthingtray fail.

> I've really meant the concrete arguments after everything is expanded, e.g. a log like

Here you go:

```
unpacking sources
unpacking source archive /nix/store/dp1xfsmxl0d8zkq35zlipbqkn5zwrr72-source
source root is source
patching sources
configuring
fixing cmake files...
cmake flags: -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON -DCMAKE_INSTALL_LOCALEDIR=/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/doc/ -DCMAKE_INSTALL_INFODIR=/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include -DCMAKE_INSTALL_SBINDIR=/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_OSX_ARCHITECTURES=x86_64 -DCMAKE_FIND_FRAMEWORK=last -DCMAKE_STRIP=/nix/store/xpzdkp9y8fnliwrs2g5pygx2qmmbrv8w-binutils-2.31.1/bin/strip -DCMAKE_RANLIB=/nix/store/xpzdkp9y8fnliwrs2g5pygx2qmmbrv8w-binutils-2.31.1/bin/ranlib -DCMAKE_AR=/nix/store/xpzdkp9y8fnliwrs2g5pygx2qmmbrv8w-binutils-2.31.1/bin/ar -DCMAKE_C_COMPILER=gcc -DCMAKE_CXX_COMPILER=g++ -DCMAKE_INSTALL_PREFIX=/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0
-- The C compiler identification is GNU 9.2.0
-- The CXX compiler identification is GNU 9.2.0
-- Check for working C compiler: /nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/gcc
-- Check for working C compiler: /nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/g++
-- Check for working CXX compiler: /nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/g++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for iconv
-- Looking for iconv - found
-- Using iconv from the standard library for target c++utilities.
-- Using __gnu_cxx::stdio_filebuf for NativeFileStream
-- Performing Test COULD_COMPILE_TEST_PROGRAM_0
-- Performing Test COULD_COMPILE_TEST_PROGRAM_0 - Success
-- Linking c++utilities against special library for std::filesystem support is not required.
-- Configuring project c++utilities
-- Using default CXX11 ABI (not forcing old CX11 ABI).
-- Using template for global.h from own (c++utilities) source directory.
-- Using template for version.h from own (c++utilities) source directory.
-- Using template for Config.cmake from own (c++utilities) source directory.
-- Using template for template.pc from own (c++utilities) source directory.
-- Could NOT find PkgConfig (missing: PKG_CONFIG_EXECUTABLE)
-- Checking for one of the modules 'cppunit>=1.13.0'
CMake Warning at cmake/modules/TestTarget.cmake:57 (message):
  Cppunit not detected via pkg-config so the version couldn't be checked.
  Required version for c++utilities is 1.13.0.
Call Stack (most recent call first):
  CMakeLists.txt:197 (include)


-- Linking test target against c++utilities
-- Using template for doxygen from own (c++utilities) source directory.
CMake Warning at cmake/modules/Doxygen.cmake:32 (message):
  Doxygen not found, unable to add target for generating API documentation
  for c++utilities
Call Stack (most recent call first):
  CMakeLists.txt:198 (include)


-- Using template for config.h from own (c++utilities) source directory.
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_FIND_USE_PACKAGE_REGISTRY
    CMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY
    CMAKE_POLICY_DEFAULT_CMP0025


-- Build files have been written to: /build/source/build
cmake: enabled parallel building
building
build flags: -j4 -l4 SHELL=/nix/store/vrnxw026hiy8jvdpaaix47x528bkzksk-bash-4.4-p23/bin/bash
Scanning dependencies of target c++utilities
[  9%] Building CXX object CMakeFiles/c++utilities.dir/application/argumentparser.cpp.o
[  9%] Building CXX object CMakeFiles/c++utilities.dir/application/fakeqtconfigarguments.cpp.o
[ 14%] Building CXX object CMakeFiles/c++utilities.dir/application/commandlineutils.cpp.o
[ 19%] Building CXX object CMakeFiles/c++utilities.dir/chrono/datetime.cpp.o
[ 23%] Building CXX object CMakeFiles/c++utilities.dir/chrono/period.cpp.o
[ 28%] Building CXX object CMakeFiles/c++utilities.dir/chrono/timespan.cpp.o
[ 33%] Building CXX object CMakeFiles/c++utilities.dir/conversion/conversionexception.cpp.o
[ 38%] Building CXX object CMakeFiles/c++utilities.dir/conversion/stringconversion.cpp.o
[ 42%] Building CXX object CMakeFiles/c++utilities.dir/io/ansiescapecodes.cpp.o
[ 47%] Building CXX object CMakeFiles/c++utilities.dir/io/binaryreader.cpp.o
[ 52%] Building CXX object CMakeFiles/c++utilities.dir/io/binarywriter.cpp.o
[ 57%] Building CXX object CMakeFiles/c++utilities.dir/io/bitreader.cpp.o
[ 61%] Building CXX object CMakeFiles/c++utilities.dir/io/inifile.cpp.o
[ 66%] Building CXX object CMakeFiles/c++utilities.dir/io/path.cpp.o
[ 71%] Building CXX object CMakeFiles/c++utilities.dir/io/nativefilestream.cpp.o
[ 76%] Building CXX object CMakeFiles/c++utilities.dir/io/misc.cpp.o
[ 80%] Building CXX object CMakeFiles/c++utilities.dir/misc/math.cpp.o
[ 85%] Building CXX object CMakeFiles/c++utilities.dir/misc/parseerror.cpp.o
[ 90%] Building CXX object CMakeFiles/c++utilities.dir/misc/levenshtein.cpp.o
[ 95%] Building CXX object CMakeFiles/c++utilities.dir/tests/testutils.cpp.o
/build/source/tests/testutils.cpp: In function 'int CppUtilities::execAppInternal(const char*, const char* const*, std::string&, std::string&, bool, int, const string&)':
/build/source/tests/testutils.cpp:375:9: warning: ignoring return value of 'int pipe(int*)', declared with attribute warn_unused_result [-Wunused-result]
  375 |     pipe(coutPipes);
      |     ~~~~^~~~~~~~~~~
/build/source/tests/testutils.cpp:376:9: warning: ignoring return value of 'int pipe(int*)', declared with attribute warn_unused_result [-Wunused-result]
  376 |     pipe(cerrPipes);
      |     ~~~~^~~~~~~~~~~
[100%] Linking CXX static library libc++utilities.a
[100%] Built target c++utilities
running tests
check flags: SHELL=/nix/store/vrnxw026hiy8jvdpaaix47x528bkzksk-bash-4.4-p23/bin/bash VERBOSE=y LD_LIBRARY_PATH=\$\(PWD\) check
/nix/store/yvq7zkvkmycdg22l4lsr4nn20174gi74-cmake-3.16.3/bin/cmake -S/build/source -B/build/source/build --check-build-system CMakeFiles/Makefile.cmake 0
make -f CMakeFiles/Makefile2 check
make[1]: Entering directory '/build/source/build'
/nix/store/yvq7zkvkmycdg22l4lsr4nn20174gi74-cmake-3.16.3/bin/cmake -S/build/source -B/build/source/build --check-build-system CMakeFiles/Makefile.cmake 0
/nix/store/yvq7zkvkmycdg22l4lsr4nn20174gi74-cmake-3.16.3/bin/cmake -E cmake_progress_start /build/source/build/CMakeFiles 30
make -f CMakeFiles/Makefile2 CMakeFiles/check.dir/all
make[2]: Entering directory '/build/source/build'
make -f CMakeFiles/c++utilities.dir/build.make CMakeFiles/c++utilities.dir/depend
make[3]: Entering directory '/build/source/build'
cd /build/source/build && /nix/store/yvq7zkvkmycdg22l4lsr4nn20174gi74-cmake-3.16.3/bin/cmake -E cmake_depends "Unix Makefiles" /build/source /build/source /build/source/build /build/source/build /build/source/build/CMakeFiles/c++utilities.dir/DependInfo.cmake --color=
make[3]: Leaving directory '/build/source/build'
make -f CMakeFiles/c++utilities.dir/build.make CMakeFiles/c++utilities.dir/build
make[3]: Entering directory '/build/source/build'
make[3]: Nothing to be done for 'CMakeFiles/c++utilities.dir/build'.
make[3]: Leaving directory '/build/source/build'
[ 70%] Built target c++utilities
make -f CMakeFiles/c++utilities_tests.dir/build.make CMakeFiles/c++utilities_tests.dir/depend
make[3]: Entering directory '/build/source/build'
cd /build/source/build && /nix/store/yvq7zkvkmycdg22l4lsr4nn20174gi74-cmake-3.16.3/bin/cmake -E cmake_depends "Unix Makefiles" /build/source /build/source /build/source/build /build/source/build /build/source/build/CMakeFiles/c++utilities_tests.dir/DependInfo.cmake --color=
Dependee "/build/source/build/CMakeFiles/c++utilities_tests.dir/DependInfo.cmake" is newer than depender "/build/source/build/CMakeFiles/c++utilities_tests.dir/depend.internal".
Dependee "/build/source/build/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/build/source/build/CMakeFiles/c++utilities_tests.dir/depend.internal".
Scanning dependencies of target c++utilities_tests
make[3]: Leaving directory '/build/source/build'
make -f CMakeFiles/c++utilities_tests.dir/build.make CMakeFiles/c++utilities_tests.dir/build
make[3]: Entering directory '/build/source/build'
[ 73%] Building CXX object CMakeFiles/c++utilities_tests.dir/tests/cppunit.cpp.o
/nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/g++  -DCPP_UTILITIES_STATIC -DCPP_UTILITIES_USE_NATIVE_FILE_BUFFER -DCPP_UTILITIES_USE_STANDARD_FILESYSTEM -I/build/source -I/build/source/build -I/build/source/..  -O3 -DNDEBUG -fvisibility=hidden   -std=gnu++17 -o CMakeFiles/c++utilities_tests.dir/tests/cppunit.cpp.o -c /build/source/tests/cppunit.cpp
[ 76%] Building CXX object CMakeFiles/c++utilities_tests.dir/tests/conversiontests.cpp.o
/nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/g++  -DCPP_UTILITIES_STATIC -DCPP_UTILITIES_USE_NATIVE_FILE_BUFFER -DCPP_UTILITIES_USE_STANDARD_FILESYSTEM -I/build/source -I/build/source/build -I/build/source/..  -O3 -DNDEBUG -fvisibility=hidden   -std=gnu++17 -o CMakeFiles/c++utilities_tests.dir/tests/conversiontests.cpp.o -c /build/source/tests/conversiontests.cpp
[ 80%] Building CXX object CMakeFiles/c++utilities_tests.dir/tests/iotests.cpp.o
/nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/g++  -DCPP_UTILITIES_STATIC -DCPP_UTILITIES_USE_NATIVE_FILE_BUFFER -DCPP_UTILITIES_USE_STANDARD_FILESYSTEM -DCPP_UTILITIES_USE_GNU_CXX_STDIO_FILEBUF -I/build/source -I/build/source/build -I/build/source/..  -O3 -DNDEBUG -fvisibility=hidden   -std=gnu++17 -o CMakeFiles/c++utilities_tests.dir/tests/iotests.cpp.o -c /build/source/tests/iotests.cpp
[ 83%] Building CXX object CMakeFiles/c++utilities_tests.dir/tests/chronotests.cpp.o
/nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/g++  -DCPP_UTILITIES_STATIC -DCPP_UTILITIES_USE_NATIVE_FILE_BUFFER -DCPP_UTILITIES_USE_STANDARD_FILESYSTEM -I/build/source -I/build/source/build -I/build/source/..  -O3 -DNDEBUG -fvisibility=hidden   -std=gnu++17 -o CMakeFiles/c++utilities_tests.dir/tests/chronotests.cpp.o -c /build/source/tests/chronotests.cpp
[ 86%] Building CXX object CMakeFiles/c++utilities_tests.dir/tests/argumentparsertests.cpp.o
/nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/g++  -DCPP_UTILITIES_STATIC -DCPP_UTILITIES_USE_NATIVE_FILE_BUFFER -DCPP_UTILITIES_USE_STANDARD_FILESYSTEM -DCPP_UTILITIES_ESCAPE_CODES_ENABLED_BY_DEFAULT -I/build/source -I/build/source/build -I/build/source/..  -O3 -DNDEBUG -fvisibility=hidden   -std=gnu++17 -o CMakeFiles/c++utilities_tests.dir/tests/argumentparsertests.cpp.o -c /build/source/tests/argumentparsertests.cpp
[ 90%] Building CXX object CMakeFiles/c++utilities_tests.dir/tests/traitstests.cpp.o
/nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/g++  -DCPP_UTILITIES_STATIC -DCPP_UTILITIES_USE_NATIVE_FILE_BUFFER -DCPP_UTILITIES_USE_STANDARD_FILESYSTEM -I/build/source -I/build/source/build -I/build/source/..  -O3 -DNDEBUG -fvisibility=hidden   -std=gnu++17 -o CMakeFiles/c++utilities_tests.dir/tests/traitstests.cpp.o -c /build/source/tests/traitstests.cpp
[ 93%] Building CXX object CMakeFiles/c++utilities_tests.dir/tests/mathtests.cpp.o
/nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/g++  -DCPP_UTILITIES_STATIC -DCPP_UTILITIES_USE_NATIVE_FILE_BUFFER -DCPP_UTILITIES_USE_STANDARD_FILESYSTEM -I/build/source -I/build/source/build -I/build/source/..  -O3 -DNDEBUG -fvisibility=hidden   -std=gnu++17 -o CMakeFiles/c++utilities_tests.dir/tests/mathtests.cpp.o -c /build/source/tests/mathtests.cpp
[ 96%] Building CXX object CMakeFiles/c++utilities_tests.dir/tests/misctests.cpp.o
/nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/g++  -DCPP_UTILITIES_STATIC -DCPP_UTILITIES_USE_NATIVE_FILE_BUFFER -DCPP_UTILITIES_USE_STANDARD_FILESYSTEM -I/build/source -I/build/source/build -I/build/source/..  -O3 -DNDEBUG -fvisibility=hidden   -std=gnu++17 -o CMakeFiles/c++utilities_tests.dir/tests/misctests.cpp.o -c /build/source/tests/misctests.cpp
[100%] Linking CXX executable c++utilities_tests
/nix/store/yvq7zkvkmycdg22l4lsr4nn20174gi74-cmake-3.16.3/bin/cmake -E cmake_link_script CMakeFiles/c++utilities_tests.dir/link.txt --verbose=y
/nix/store/fqhjxf9ii4w4gqcsx59fyw2vvj91486a-gcc-wrapper-9.2.0/bin/g++  -O3 -DNDEBUG  -rdynamic CMakeFiles/c++utilities_tests.dir/tests/cppunit.cpp.o CMakeFiles/c++utilities_tests.dir/tests/conversiontests.cpp.o CMakeFiles/c++utilities_tests.dir/tests/iotests.cpp.o CMakeFiles/c++utilities_tests.dir/tests/chronotests.cpp.o CMakeFiles/c++utilities_tests.dir/tests/argumentparsertests.cpp.o CMakeFiles/c++utilities_tests.dir/tests/traitstests.cpp.o CMakeFiles/c++utilities_tests.dir/tests/mathtests.cpp.o CMakeFiles/c++utilities_tests.dir/tests/misctests.cpp.o  -o c++utilities_tests  /nix/store/pnbawf706qj3n98kwfbfdjlk2iwin5wf-cppunit-1.15.0/lib/libcppunit.so libc++utilities.a
make[3]: Leaving directory '/build/source/build'
[100%] Built target c++utilities_tests
make -f CMakeFiles/check.dir/build.make CMakeFiles/check.dir/depend
make[3]: Entering directory '/build/source/build'
cd /build/source/build && /nix/store/yvq7zkvkmycdg22l4lsr4nn20174gi74-cmake-3.16.3/bin/cmake -E cmake_depends "Unix Makefiles" /build/source /build/source /build/source/build /build/source/build /build/source/build/CMakeFiles/check.dir/DependInfo.cmake --color=
Dependee "/build/source/build/CMakeFiles/check.dir/DependInfo.cmake" is newer than depender "/build/source/build/CMakeFiles/check.dir/depend.internal".
Dependee "/build/source/build/CMakeFiles/CMakeDirectoryInformation.cmake" is newer than depender "/build/source/build/CMakeFiles/check.dir/depend.internal".
Scanning dependencies of target check
make[3]: Leaving directory '/build/source/build'
make -f CMakeFiles/check.dir/build.make CMakeFiles/check.dir/build
make[3]: Entering directory '/build/source/build'
/nix/store/yvq7zkvkmycdg22l4lsr4nn20174gi74-cmake-3.16.3/bin/ctest -V
UpdateCTestConfiguration  from :/build/source/build/DartConfiguration.tcl
UpdateCTestConfiguration  from :/build/source/build/DartConfiguration.tcl
Test project /build/source/build
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 1
    Start 1: c++utilities_run_tests

1: Test command: /build/source/build/c++utilities_tests "-p" "/build/source/testfiles" "-w" "/build/source/build/testworkingdir"
1: Test timeout computed to be: 10000000
1: /build/source/testfiles/
1: /build/source/testfiles/
1: ./testfiles/
1: Directory used to store working copies:
1: /build/source/build/testworkingdir/
1: Executing test cases ...
1: ................
1: output for formatting with ANSI escape codes:
1: Error: some error
1: Warning: some warning
1: Info: some info
1: ==> ERROR: Arch-style error
1: ==> WARNING: Arch-style warning
1:     Arch-style message
1: ==> Arch-style success
1:   -> Arch-style sub-message
1: blue, blinking text on red background
1: ---------------------------------------------
1: .......................
1:
1:
1: OK (39 tests)
1:
1:
1: Tests successful
1/1 Test #1: c++utilities_run_tests ...........   Passed    0.01 sec

100% tests passed, 0 tests failed out of 1

Total Test time (real) =   0.01 sec
make[3]: Leaving directory '/build/source/build'
[100%] Built target check
make[2]: Leaving directory '/build/source/build'
/nix/store/yvq7zkvkmycdg22l4lsr4nn20174gi74-cmake-3.16.3/bin/cmake -E cmake_progress_start /build/source/build/CMakeFiles 0
make[1]: Leaving directory '/build/source/build'
installing
install flags: SHELL=/nix/store/vrnxw026hiy8jvdpaaix47x528bkzksk-bash-4.4-p23/bin/bash install
[100%] Built target c++utilities
Install the project...
-- Install configuration: "Release"
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/tests/calculateoverallcoverage.awk
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/coding-style.clang-format
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/c++utilitiesConfig.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/c++utilitiesConfigVersion.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/lib/pkgconfig/c++utilities.pc
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/lib/libc++utilities.a
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/c++utilitiesTargets.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/c++utilitiesTargets-release.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/application/argumentparser.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/application/commandlineutils.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/application/fakeqtconfigarguments.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/application/global.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/chrono/datetime.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/chrono/period.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/chrono/timespan.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/chrono/format.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/conversion/binaryconversion.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/conversion/binaryconversionprivate.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/conversion/conversionexception.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/conversion/stringconversion.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/conversion/stringbuilder.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/io/ansiescapecodes.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/io/binaryreader.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/io/binarywriter.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/io/bitreader.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/io/copy.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/io/inifile.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/io/path.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/io/nativefilestream.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/io/misc.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/misc/flagenumclass.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/misc/math.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/misc/multiarray.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/misc/parseerror.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/misc/traits.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/misc/levenshtein.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/tests/testutils.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/tests/cppunit.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/tests/outputcheck.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/global.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/include/c++utilities/version.h
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/modules/BasicConfig.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/modules/ConfigHeader.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/modules/LibraryTarget.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/modules/TestTarget.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/modules/AppTarget.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/modules/WindowsResources.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/modules/TemplateFinder.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/modules/Doxygen.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/modules/ListToString.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/modules/ShellCompletion.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/modules/3rdParty.cmake
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/templates/bash-completion.sh.in
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/templates/Config.cmake.in
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/templates/config.h.in
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/templates/desktop.in
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/templates/appdata.xml.in
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/templates/doxygen.in
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/templates/global.h.in
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/templates/version.h.in
-- Installing: /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/share/c++utilities/cmake/templates/template.pc.in
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0
strip is /nix/store/xpzdkp9y8fnliwrs2g5pygx2qmmbrv8w-binutils-2.31.1/bin/strip
stripping (with command strip and flags -S) in /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0/lib
patching script interpreter paths in /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0
checking for references to /build/ in /nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0...
/nix/store/ghv3qk4by8fsgimwwg7dr358gbh6zfxy-cpp-utilities-5.2.0
```

> And is the error only in the config file or are the headers installed into the wrongly prefixed directory, too?

The files themselves are installed fine. It's just this very line that seems wrong.

--- a/pkgs/development/libraries/cpp-utilities/default.nix
+++ b/pkgs/development/libraries/cpp-utilities/default.nix
@@ -1,4 +1,9 @@
-{ stdenv, fetchFromGitHub, cmake, cppunit }:
+{ stdenv
+, fetchFromGitHub
+, fetchpatch
+, cmake
+, cppunit
+}:
 
 stdenv.mkDerivation rec {
   pname = "cpp-utilities";
@@ -10,6 +15,17 @@ stdenv.mkDerivation rec {
     rev = "v${version}";
     sha256 = "0cghk1a0ki1063ci63imakmggwzkky1hx6lhrvc0wjfv754wsklb";
   };
+  # See https://github.com/Martchus/cpp-utilities/issues/18
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/Martchus/cpp-utilities/commit/b2a2773cdfb2b0017a3fa3d0ed2259a9a5fda666.patch";
+      sha256 = "01js90ba4xxljifncm48zbxmg7mwwz1gla1hn87yzbic47d85hfj";
+    })
+    (fetchpatch {
+      url = "https://github.com/Martchus/cpp-utilities/commit/4dd2179f191d1ace113f26177944684fa1561dc1.patch";
+      sha256 = "0chw33mwsvj7cigd1c4xl2zhpbfsp5rrijdm46qpn78bq70xcz9j";
+    })
+  ];
 
   nativeBuildInputs = [ cmake ];
   checkInputs = [ cppunit ];

